src/pudl/transform/ferc1.py

```diff
@@ -1188,11 +1196,15 @@ class PlantInServiceFerc1TableTransformer(Ferc1AbstractTableTransformer):
     @cache_df(key="dbf")
     def align_row_numbers_dbf(self, df: pd.DataFrame) -> pd.DataFrame:
         """Align historical FERC1 DBF row numbers with XBRL account IDs."""
-        return pd.merge(
-            df,
-            get_ferc1_dbf_xbrl_glue(self.source_table_id(Ferc1Source.DBF)),
-            on=["report_year", "row_number"],
-        ).rename(columns={"xbrl_column_stem": "ferc_account_label"})
+        dbf_to_xbrl_map = fill_dbf_to_xbrl_map(
+            df=read_dbf_to_xbrl_map(
+                dbf_table_name=self.source_table_id(Ferc1Source.DBF)
+            ),
+            dbf_years=Ferc1Settings().dbf_years,
```

Comment:
is always including all of the dbf_years always what we want? i assume this process isn't super intensive, but will this cause unintended consequences?

Reply:
Hmm. Thinking about this a little big, the years are fragile. Like only certain sets of years can be expected to work with any given mapping. I've added some defensive checks to the filling function.

I don't think that this will cause problems, or if it does, I don't think this is the place to address them. Every real DBF-XBRL mapping **should cover all of the years of data**, and extra years of mappings which aren't applicable to the data being processed will get excluded in the merge since they don't match to any of the `(report_year, row_number)` tuples in the dataframe.

I added the years as a parameter because specifying all the years from 1994-2020 for testing seemed dumb.

I'll make all the FERC1 years the default parameter so it's not tempting for people to mess with it.

Reply:
Whoops need to remove the param here since it's the default.

Reply:
Removed.